api/v1alpha1/ipfscluster_types.go

```diff
@@ -60,28 +71,29 @@ type followParams struct {
 	Template string `json:"template"`
 }
 
-// networkConfig defines the configuration structure used for networking.
-type networkConfig struct {
+// NetworkConfig defines the configuration structure used for networking.
+type NetworkConfig struct {
+	// circuitRelays defines how many CircuitRelays should be created.
 	CircuitRelays int32 `json:"circuitRelays"`
+	// networkMode is a switch which defines whether this IPFSCluster will use
+	// the global IPFS network or create its own.
+	// +kubebuilder:validation:Enum={public,private}
+	NetworkMode NetworkMode `json:"networkMode,omitempty"`
```

Comment:
Maybe this could just be the `Public` boolean flag from the spec instead of an enum

Reply:
+1